server/src/models/message.model.ts

```diff
@@ -0,0 +1,22 @@
+import mongoose, { Schema, Types } from 'mongoose';
+
+interface IMessageModel {
+	text: string;
+	room_id: Types.ObjectId;
+	user_email: string;
+	created_at: Date;
+	updated_at: Date;
+}
+
+const messageSchema = new Schema<IMessageModel>(
+	{
+		text: String,
+		room_id: Schema.Types.ObjectId,
+		user_email: String,
+	},
+	{
+		timestamps: true,
+	}
+);
```

Comment:
please fix that to my suggestion - 
some of them optionals please remove unnecessary fields.

```suggestion
const messageSchema = new mongoose.Schema({
    text: {
        type: String,
        required: true,  // The message content
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',     // The user whom sent the message
    },
    room: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Room',     // The room where the message was sent
    },
    sentAt: {
        type: Date,
        default: Date.now, // Timestamp of when the message was sent
    },
    isRead: {
        type: Boolean,
        default: false, // For tracking if a message has been read by the recipient(s)
    },
    attachments: [
        {
            url: String,        // URL of an attachment (image, video, etc.)
            type: String,       // Type of attachment: image, video, file, etc.
        },
    ],
    reactions: [
        {
            user: { 
                type: mongoose.Schema.Types.ObjectId, 
                ref: 'User' 
            },
            reactionType: { 
                type: String, 
                enum: ['like', 'love', 'laugh', 'sad', 'angry'], 
                default: 'like' 
            },
        },
    ],
}, { timestamps: true });

```